tensorflow_lite_support/ios/task/vision/sources/TFLImageClassifier.m

```diff
@@ -0,0 +1,236 @@
+    self.baseOptions.modelFile.filePath = modelPath;
+    self.baseOptions.computeSettings.cpuSettings.numThreads = -1;
+    self.classificationOptions = [[TFLClassificationOptions alloc] init];
+    self.classificationOptions.maxResults = -1;
+  }
+  return self;
+}
+@end
+
+@implementation TFLImageClassifier
+- (void)dealloc {
+  TfLiteImageClassifierDelete(_imageClassifier);
+}
+
++ (TFLImageClassifierOptions *)defaultImageClassifierOptionsWithModelPath:
+    (nonnull NSString *)modelPath {
+  TFLImageClassifierOptions *options =
+      [[TFLImageClassifierOptions alloc] initWithModelPath:modelPath];
+  if (!options) return nil;
+
+  options.classificationOptions.maxResults = 5;
+  options.classificationOptions.scoreThreshold = 0;
+
+  return options;
+}
+
++ (void)deleteCStringsArray:(char **)cStrings count:(int)count {
+  for (NSInteger i = 0; i < count; i++) {
+    free(cStrings[i]);
+  }
+
+  free(cStrings);
+}
+
++ (char **)cStringArrayFromNSArray:(NSArray<NSString *> *)strings error:(NSError **)error {
+  if (strings.count <= 0) {
+    [TFLCommonUtils customErrorWithCode:TFLSupportErrorCodeInvalidArgumentError
+                            description:@"Invalid length of strings found for list type options."
+                                  error:error];
+    return NULL;
+  }
+
+  char **cStrings = (char **)calloc(strings.count, sizeof(char *));
+
+  if (!cStrings) {
+    [TFLCommonUtils customErrorWithCode:TFLSupportErrorCodeInternalError
+                            description:@"Could not initialize list type options."
+                                  error:error];
+    return nil;
+  }
+
+  for (NSInteger i = 0; i < strings.count; i++) {
+    char *cString = [TFLCommonUtils
+        mallocWithSize:[strings[i] lengthOfBytesUsingEncoding:NSUTF8StringEncoding] + 1
+                 error:error];
+    if (!cString) return nil;
+
+    strcpy(cString, strings[i].UTF8String);
+  }
+
+  return cStrings;
+}
+
++ (nullable instancetype)imageClassifierWithModelPath:(nonnull NSString *)modelPath
+                                                error:(NSError **)error {
+  TFLImageClassifierOptions *options = [self defaultImageClassifierOptionsWithModelPath:modelPath];
+
+  TFLImageClassifier *imageClassifier = nil;
+  if (options)
+    imageClassifier = [TFLImageClassifier imageClassifierWithOptions:options error:error];
+  else
+    [TFLCommonUtils
+        customErrorWithCode:TFLSupportErrorCodeInternalError
+                description:@"Some error occured during initialization of Image Classifier."
+                      error:error];
+
+  return imageClassifier;
+}
+
++ (nullable instancetype)imageClassifierWithOptions:(nonnull TFLImageClassifierOptions *)options
+                                              error:(NSError **)error {
+  TfLiteImageClassifierOptions cOptions = TfLiteImageClassifierOptionsCreate();
+
+  cOptions.classification_options.score_threshold = options.classificationOptions.scoreThreshold;
```

Comment:
To avoid having to repeat this copy logic in all tasks that involve classification, what about implementing a objC->C copy method in the TFLImageClassifierOptions and TFLBaseOptions class? You can put it in an extension (e.g. TFLImageClassifierOptions+C.h) to avoid exposing the API to the ObjC users.

Reply:
I have added categories for base options and classification options and moved the operations to them.